module/zfs/arc.c

```diff
@@ -3312,7 +3313,10 @@ arc_hdr_realloc_crypt(arc_buf_hdr_t *hdr, boolean_t need_crypt)
 		buf->b_hdr = nhdr;
 		mutex_exit(&buf->b_evict_lock);
 	}
-
+	if (need_transfer)
+		zfs_refcount_transfer_ownership_many(
+		    &hdr->b_l1hdr.b_state->arcs_size,
+		    arc_hdr_size(hdr), hdr, nhdr);
```

Comment:
it would be nice to use zfs_refcount_transfer_ownership_many() in arc_hdr_realloc() as well.